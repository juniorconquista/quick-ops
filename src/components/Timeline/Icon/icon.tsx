import React, { memo } from 'react';
import { variant, VariantArgs } from 'styled-system';
import styled from 'styled-components';

import { IconStyledProps } from './icon.types';

const variants = variant({
    prop: 'styling',
    variants: {
        base: {
            height: '59px',
            width: '59px',
            borderRadius: '50%',
            display: 'flex',
            alignItems: 'center',
            justifyContent: 'center',
            bg: '#f0f3f7',
            mb: '10px',
            svg: {
                height: '25px',
                width: '25px',
                fill: '#6b7480',
            },
        },
        get 'hotel-fee'() {
            const base: { [key: string]: any } = this.base;
            return {
                ...base,
                bg: '#e7f2fd',
                svg: {
                    ...base.svg,
                    fill: '#0f68bd',
                },
            };
        },
        get food() {
            const base: { [key: string]: any } = this.base;
            return {
                ...base,
                bg: '#fde7e7',
                svg: {
                    ...base.svg,
                    fill: '#bd0f0f',
                },
            };
        },
        get EVALUATION() {
            const base: { [key: string]: any } = this.base;
            return {
                ...base,
                bg: '#f3e7fd',
                svg: {
                    ...base.svg,
                    fill: '#660fbd',
                },
            };
        },
        get ACCOUNTABILITY_SUBMITTED() {
            const base: { [key: string]: any } = this.base;
            return {
                ...base,
                bg: '#ecf8f8',
                svg: {
                    ...base.svg,
                    height: '20px',
                    width: '20px',
                    fill: '#03a046',
                },
            };
        },
        get ACCOUNTABILITY_CREATED() {
            const base: { [key: string]: any } = this.base;
            return {
                ...base,
                bg: '#fdf8e7',
                svg: {
                    ...base.svg,
                    height: '20px',
                    width: '20px',
                    fill: '#bd6b0f',
                },
            };
        },
    },
} as VariantArgs);

const Icon: React.FC<IconStyledProps> = styled.div<IconStyledProps>`
    ${variants}
`;

export default memo(Icon);
