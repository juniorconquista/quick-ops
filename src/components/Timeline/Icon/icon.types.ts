import { ReactNode } from 'react';

export type IconStyledProps = {
    /** Each styling variant has unique properties that are inherited through the type informed */
    styling?:
        | 'base'
        | 'hotel-fee'
        | 'EVALUATION'
        | 'ACCOUNTABILITY_SUBMITTED'
        | 'ACCOUNTABILITY_CREATED';
    children?: ReactNode;
};
