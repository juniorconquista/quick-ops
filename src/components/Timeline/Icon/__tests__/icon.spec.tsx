import React from 'react';
import renderer from 'react-test-renderer';
import { cleanup, render } from '@testing-library/react';
import ThemeProvider from '../../../../utils/ThemeProviderTest';

import 'jest-styled-components';
import '@testing-library/jest-dom/extend-expect';

import Icon from '../icon';

describe('Icon component', () => {
    afterEach(() => {
        jest.clearAllMocks();
        cleanup();
    });
    test('basic snapshot', () => {
        const container = render(
            <ThemeProvider>
                <Icon styling="base" />
            </ThemeProvider>,
        );
        expect(container.asFragment()).toMatchSnapshot();
    });

    it('should inherit the styling properties of the hotel-fee variant', () => {
        const icon = renderer
            .create(
                <ThemeProvider>
                    <Icon styling="hotel-fee" />
                </ThemeProvider>,
            )
            .toJSON();
        expect(icon).toHaveStyleRule('background-color', '#e7f2fd');
        expect(icon).toHaveStyleRule('fill', '#0f68bd', {
            modifier: 'svg',
        });
    });

    it('should inherit the styling properties of the food variant', () => {
        const icon = renderer
            .create(
                <ThemeProvider>
                    <Icon styling="food" />
                </ThemeProvider>,
            )
            .toJSON();
        expect(icon).toHaveStyleRule('background-color', '#fde7e7');
        expect(icon).toHaveStyleRule('fill', '#bd0f0f', {
            modifier: 'svg',
        });
    });

    it('should inherit the styling properties of EVALUATION variant', () => {
        const icon = renderer
            .create(
                <ThemeProvider>
                    <Icon styling="EVALUATION" />
                </ThemeProvider>,
            )
            .toJSON();
        expect(icon).toHaveStyleRule('background-color', '#f3e7fd');
        expect(icon).toHaveStyleRule('fill', '#660fbd', {
            modifier: 'svg',
        });
    });

    it('should inherit the styling properties of ACCOUNTABILITY_SUBMITTED variant', () => {
        const icon = renderer
            .create(
                <ThemeProvider>
                    <Icon styling="ACCOUNTABILITY_SUBMITTED" />
                </ThemeProvider>,
            )
            .toJSON();
        expect(icon).toHaveStyleRule('background-color', '#ecf8f8');
        expect(icon).toHaveStyleRule('fill', '#03a046', {
            modifier: 'svg',
        });
        expect(icon).toHaveStyleRule('width', '20px', {
            modifier: 'svg',
        });
        expect(icon).toHaveStyleRule('height', '20px', {
            modifier: 'svg',
        });
    });

    it('should inherit the styling properties of ACCOUNTABILITY_CREATED variant', () => {
        const icon = renderer
            .create(
                <ThemeProvider>
                    <Icon styling="ACCOUNTABILITY_CREATED" />
                </ThemeProvider>,
            )
            .toJSON();
        expect(icon).toHaveStyleRule('background-color', '#fdf8e7');
        expect(icon).toHaveStyleRule('fill', '#bd6b0f', {
            modifier: 'svg',
        });
        expect(icon).toHaveStyleRule('width', '20px', {
            modifier: 'svg',
        });
        expect(icon).toHaveStyleRule('height', '20px', {
            modifier: 'svg',
        });
    });
});
