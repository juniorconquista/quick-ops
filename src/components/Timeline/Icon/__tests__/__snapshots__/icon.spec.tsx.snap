// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component basic snapshot 1`] = `
<DocumentFragment>
  .c0 {
  height: 59px;
  width: 59px;
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #f0f3f7;
  margin-bottom: 10px;
}

.c0 svg {
  height: 25px;
  width: 25px;
  fill: #6b7480;
}

<div
    class="c0"
  />
</DocumentFragment>
`;
